Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Middlewares/AWSS3ErrorWith200StatusXMLMiddleware.swift

```diff
@@ -40,9 +35,43 @@ extension AWSS3ErrorWith200StatusXMLMiddleware: HttpInterceptor {
         context: some MutableResponse<Self.InputType, Self.RequestType, Self.ResponseType>
     ) async throws {
         let response = context.getResponse()
-        if try await isErrorWith200Status(response: response) {
-            response.statusCode = errorStatusCode
-            context.updateResponse(updated: response)
+
+        // Check if the status code is OK (200)
+        guard response.statusCode == .ok else {
+            return
+        }
+
+        // Check if the response body contains an XML Error
+        guard let data = try await response.body.readData() else {
+            return
+        }
+
+        // Determine if there is an error in result (500) or keep status (200)
+        let statusCode = try isRootErrorElement(data: data) ? errorStatusCode : response.statusCode
+
+        // For event streams the body needs to be copied as buffered streams are non-seekable
+        let updatedBody = response.body.copy(data: data)
+
+        let updatedResponse = response.copy(
+            body: updatedBody,
+            statusCode: statusCode
+        )
```

Comment:
Q: Am I understanding correctly: since the body can't be read twice, the read body is copied back in to the response for it to be available down the line in operation stack?

Reply:
Yes otherwise it will be `nil` when the user tries to look at the events